mirascope/chat/gemini/prompt.py

```diff
@@ -0,0 +1,155 @@
+"""A module for prompting Google's Gemini Chat API."""
+import datetime
+import logging
+import re
+from inspect import isclass
+from typing import Annotated, ClassVar, Generator, Optional, Type, TypeVar
+
+from google.generativeai import GenerativeModel, configure  # type: ignore
+from google.generativeai.types import ContentsType  # type: ignore
+from pydantic import AfterValidator, BaseModel, ValidationError
+
+from ...prompts import Prompt, tool_fn
+from ...prompts.prompt import format_template
+from .tools import GeminiTool
+from .types import GeminiCallParams, GeminiCompletion, GeminiCompletionChunk
+
+logger = logging.getLogger("mirascope")
+BaseModelT = TypeVar("BaseModelT", bound=BaseModel)
+
+
+class GeminiPrompt(Prompt):
```

Comment:
will there be an async prompt?

Reply:
I think just `async_create` etc. functions?